libtransmission/rpcimpl.cc

```diff
@@ -2332,141 +2054,634 @@ void add_strings_from_var(std::set<std::string_view>& strings, tr_variant const&
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_pex_enabled,
+        [](tr_session const& src) -> tr_variant { return src.allows_pex(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<bool>())
+            {
+                tr_sessionSetPexEnabled(&tgt, *val);
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_port_forwarding_enabled,
+        [](tr_session const& src) -> tr_variant { return tr_sessionIsPortForwardingEnabled(&src); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<bool>())
+            {
+                tr_sessionSetPortForwardingEnabled(&tgt, *val);
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_preferred_transports,
+        [](tr_session const& src) -> tr_variant { return src.save_preferred_transports(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& err)
+        {
+            if (!tgt.load_preferred_transports(src))
+            {
+                err = { JsonRpc::Error::INVALID_PARAMS, R"(the list must be unique with the values "utp" or "tcp")" };
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_queue_stalled_enabled,
+        [](tr_session const& src) -> tr_variant { return src.queueStalledEnabled(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<bool>())
+            {
+                tr_sessionSetQueueStalledEnabled(&tgt, *val);
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_queue_stalled_minutes,
+        [](tr_session const& src) -> tr_variant { return src.queueStalledMinutes(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<int64_t>())
+            {
+                tr_sessionSetQueueStalledMinutes(&tgt, static_cast<int>(*val));
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_rename_partial_files,
+        [](tr_session const& src) -> tr_variant { return src.isIncompleteFileNamingEnabled(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<bool>())
+            {
+                tr_sessionSetIncompleteFileNamingEnabled(&tgt, *val);
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_reqq,
+        [](tr_session const& src) -> tr_variant { return src.reqq(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<int64_t>())
+            {
+                tgt.set_reqq(*val);
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_rpc_version,
+        [](tr_session const& /*src*/) -> tr_variant { return tr_variant::unmanaged_string(RpcVersion); },
```

Comment:
We probably should make `tr_quark` enum class instead of a typedef in a future PR. I almost didn't spot this.

```suggestion
        [](tr_session const& /*src*/) -> tr_variant { return RpcVersion; },
```